go/consensus/tendermint/apps/roothash/transactions.go

```diff
@@ -119,7 +121,11 @@ func (app *rootHashApplication) executorCommit(
 	if err != nil {
 		return err
 	}
-	defer state.SetRuntimeState(rtState)
+	defer func() {
+		if err2 := state.SetRuntimeState(ctx, rtState); err2 != nil {
```

Comment:
I think one idea was that since commitments are independently verified, we should still add all that are valid, even if some are not. However, we currently fail on the first error with executor commitments, so this would only be applicable to merge commitments.

So on second thought I've now changed this to only commit runtime state in case all commitments were valid and there were no other errors.